### PR TITLE
feat: reconnect all & create-invitation in mediator command

### LINKED ISSUE #2328: reconnect all and create-invitation features in mediator commands

### PR DESCRIPTION
**Pull request by sudeshrshetty**
- also added REST & JS bindings
- closes #2328

TODO: mobile bindings

Signed-off-by: sudesh.shetty <sudesh.shetty@securekey.com>

### CODE REVIEW

**Review by fqutishat — APPROVED** [review]

File: pkg/controller/command/mediator/command.go [hunk 1]
```diff
@@ -244,6 +269,74 @@ func (o *Command) Reconnect(rw io.Writer, req io.Reader) command.Error {
...
+
+	logutil.LogDebug(logger, CommandName, ReconnectAllCommandMethod, successString)
+
+	return nil
+}
+
+// CreateInvitation creates out-of-band invitation from one of the mediator connections.
+func (o *Command) CreateInvitation(rw io.Writer, req io.Reader) command.Error {
+	connections, err := o.routeClient.GetConnections()
+	if err != nil {
```
Comment (soluchok) [inline pkg/controller/command/mediator/command.go:301]: I see that this function only adds a random connection. I don't think that we need a new function for that.
I would suggest modifying the existing function (outOfBand.CreateInvitation) to support random `router connection`. 
To be honest, I don't like this random pick up at all. I think it should be part of edge-agent refactoring. After we set up the router we have a connection, so we can use it everywhere we need it.

@sudeshrshetty @llorllale @rolsonquadras ^

Comment (sudeshrshetty) [reply pkg/controller/command/mediator/command.go:301]: Nothing wrong in adding a command to reduce number of lines or reduce repeated logic for clients. This is moved from agent-sdk.

if you pass all connections then out of band client picks first one always, which is not what we need for router.

As comment says, this command is for clients who can directly create invitation without passing any connection details, like `mediator.createInvitation()`.  Mediator will pickup one random connection (current webwallet logic).

If someone wants to create invitation for particular connection, he doesn't need this command.

Comment (llorllale) [reply pkg/controller/command/mediator/command.go:301]: I agree the randomization of the router selection should be exposed in the clients and not implemented here.

Comment (llorllale) [reply pkg/controller/command/mediator/command.go:301]: We should not be reimplementing this logic everywhere.

Comment (sudeshrshetty) [reply pkg/controller/command/mediator/command.go]: removed `mediator.createInvitation()`

**Review by soluchok — APPROVED** [review]

**Review by rolsonquadras — APPROVED** [review]

**Pull request merged**